backend/app/database/database_service.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+from app.database.results import CommandResult, QueryResult
+from app.models import TableQuery
+
+
+class DatabaseService(ABC):
+    @abstractmethod
+    def command(self, table_query: TableQuery) -> CommandResult:
+        pass
+
+    @abstractmethod
+    def query(self, table_query: TableQuery) -> QueryResult:
```

Comment:
Instead of passing strings, like you had before, you're passing a structured version of the string. That misses the point. This service should work on the level of high-level models like User, not on the level of queries.